src/components/course-administration/add-student.cjsx

```diff
@@ -0,0 +1,80 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+{RosterActions, RosterStore} = require '../../flux/roster'
+{TutorInput} = require '../tutor-input'
+BindStoreMixin = require '../bind-store-mixin'
+
+BLANK_STUDENT = first_name: '', last_name: '', email: '', password: ''
+
+Field = React.createClass
+  propTypes:
+    label: React.PropTypes.string.isRequired
+    name:  React.PropTypes.string.isRequired
+    value: React.PropTypes.string.isRequired
+    onChange:  React.PropTypes.func.isRequired
+    autofocus: React.PropTypes.bool
+
+  componentDidMount: ->
+    @refs.input.focus() if @props.autofocus
+
+  onChange: (value) ->
+    @props.onChange(value)
+
+  render: ->
+    <TutorInput
+      ref="input"
+      label={@props.label}
+      value={@props.value}
+      required={true}
+      onChange={@onChange} />
+
+module.exports = React.createClass
+  displayName: 'AddStudentLink'
+  propTypes:
+    courseId: React.PropTypes.string.isRequired
+    period: React.PropTypes.object.isRequired
+
+  mixins: [BindStoreMixin]
+  bindStore: RosterStore
+  bindEvent: 'created'
+
+  getInitialState: ->
+    BLANK_STUDENT
```

Comment:
No really because React will clone anything that's passed into state.  However, I don't think that's documented or anything so they might change the behavior at some point.  I'll add a `_.clone`